src/modules/sub-items/sub.items.module.js

```diff
@@ -240,6 +304,57 @@ export default class SubItemsModule extends Component {
         }, {});
     }
 
+    updateTotalCountState(totalCount) {
+        this.setState(() => ({
+            totalCount,
+        }));
+    }
+
+    discardAllPagesOtherThan(page) {
+        this.updateListViewHeight();
+        this.setState((state) => ({
+            pages: {
+                [page]: state.pages[page],
+            },
+        }));
+    }
+
+    discardAllPages() {
+        this.updateListViewHeight();
+        this.setState(() => ({
+            pages: {},
+        }));
+    }
+
+    discardCurrentPage() {
+        const { currentPage } = this.state;
+
+        this.updateListViewHeight();
+        this.setState((state) => ({
+            pages: {
+                ...state.pages,
+                [currentPage]: null,
+            },
+        }));
+    }
+
+    changeSorting(sortClause) {
+        this.updateListViewHeight();
+        this.setState((state) => ({
+            sortClause,
+            sortOrder: state.sortClause === sortClause ? this.getOppositeSortOrder(state.sortOrder) : this.getDefaultSortOrder(),
```

Comment:
I would expose it as a class method: `getSortOrder`: `state.sortClause === sortClause ? this.getOppositeSortOrder(state.sortOrder) : this.getDefaultSortOrder()`

What do you think about this approach?

Reply:
This was my first approach but `getSortOrder` is not very descriptive and you have to pass `state.sortClause`, `state.sortOrder` and `sortClause` to this method which for me obscures what it does.
But if you prefer `getSortOrder` I can change it. :)

Reply:
I think, `getSortOrder` makes sense.